src/services/WalletService.ts

```
@@ -129,6 +133,12 @@ export default abstract class WalletService {
     const maxAddressPerCall: number = this.getWalletAddressesPerCall();
     let addresses: WalletAddress[] = [];
     try {
+      const connected = await this.isConnected();
+
+      if (!connected) {
+        await this.reconnect();
```

we should probably raise an exception if we cannot reconnect